### ISSUE TITLE
140717 Not Playing 3+

### ISSUE THREAD

**Issue by Truthdoc** [ddugovic/Stockfish#423]
Hi guys (FischyVishy here)

I was using the 140717 version of SF to analyze a 3+ opening, and I noticed that its evaluation was in stark contrast to that of the SF on lichess; I tried it with the 301216 version, and it agreed with the one on lichess.

The line was 1. e4 g6 2. d4 Bg7 3. Nc3 c6 4. Nf3 d5 5. Nxd5

Here are some screenshots:

https://imgur.com/a/VFJ4O
https://imgur.com/a/JhsMK
https://imgur.com/a/CVxB1

Basically the 140717 version doesn't seem to be playing 3+, even though in the screenshots you can see that I've marked the 3+ option. Niklas couldn't replicate it, so I was wondering if maybe 140717 just isn't playing 3+ or if there's something weird going on with my comp.

**Comment by ddugovic** [comment]
At some point the variant name changed from `threecheck` to `3check`.  It would be interesting to see the GUI / engine communication log.

**Comment by ianfab** [comment]
[UCI2WB converts `3check` to `threecheck`](http://hgm.nubati.net/cgi-bin/gitweb.cgi?p=uci2wb.git;a=blob;f=UCI2WB.c;h=bf72b3463fedc149ee9ca311e43b12f2df18062c;hb=HEAD#l560) as far as I can see from the code, and if I remember correctly multi-variant Stockfish was the main reason for this conversion.

@HGMuller Are you aware of any engines that still use the spelling `threecheck`? If not, could the behavior of UCI2WB be changed for 3check, since I think it would anyway simplify the code and make UCI and CECP more consistent in that regard?

**Comment by ddugovic** [comment]
@Truthdoc says he is using WinBoard version 4.9.170105.

**Comment by HGMuller** [comment]
Ah, I did miss this. Yes, I can change that in UCI2WB. The name conversion was done entirely for the benefit of Stockfish-variant; there is no other UCI engine I know that plays this variant. So I don't expect there would be any compatibility problems.
I did not want to force CECP variant names on UCI; they are not that great. (Like 'fischerandom' with a single 'r'...) OTOH, it certainly simplifies life when the names are the same. Another issue is how the should look in the Variant tag of PGN, and if UCI and CECP names are the same, the obvious choice would be to make that the PGN name too. I suppose most existing PGN for 3check games would use the CECP name, as they were likely created by people using XBoard to play 3check on ICC or FICS.

**Comment by ianfab** [comment]
@HGMuller Thanks. I do not have a strong preference regarding the tag in the PGN, but I agree that it would make sense to use `3check` for consistency.

**Comment by HGMuller** [comment]
So what exactly do you expect me to do? The patch of UCI2WB to refrain from translating 3check back and forth to threecheck was pushed to the hgm.nubati.net repository 2 days ago. I admit I haven't tested it, but it seemed trivial enough to not need that.

**Comment by ianfab** [comment]
@HGMuller I think that your [most recent commit for UCI2WB](http://hgm.nubati.net/cgi-bin/gitweb.cgi?p=uci2wb.git;a=commit;h=a7299318dbfbb4ead5ed39d2ea8c586bc79ad182) should be sufficient to fix this issue.

@Truthdoc Could you please test whether it is working after the update of UCI2WB? If it does, please close this issue.

**Comment by Truthdoc** [comment]
@ianfab I cannot access that link.

**Comment by HGMuller** [comment]
For some reason the GitWeb script for browsing my on-line source repository has stopped working last month. Probably because the provider updated Perl, and the GitWeb script is no longer compatible with the new Perl version. I have not figured out how to cure this yet. (I don't know Perl at all.)

In the mean time you can use my backup repository at [](http://winboard.nl/cgi-bin?p=uci2wb.git;a=summary) , which I also updated for UCI2WB.

**Comment by ianfab** [comment]
Since there has not been any report that it is not working after the fix, I am closing this.

**Comment by ddugovic** [comment]
@Truthdoc has asked me from where one can download the latest WinBoard executable; I think I found it at http://www.open-aurec.com/wbforum/viewtopic.php?t=51528 and it appears to bundle a SF which plays variants, but I couldn't figure out how to start a 3+ game.

**Comment by HGMuller** [comment]
The version that bundles Stockfish-variant is at 
http://hgm.nubati.net/WinBoard-AA.zip . It is a version of WinBoard 4.9 
that never made it beyond the beta stage, because I was too busy with 
other things, an the official GNU XBoard project is currently without an 
admin.


Op 6/5/2018 om 1:12 PM schreef Daniel Dugovic:
